src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/codeSmell/ParameterEqualsDefaultValueInspector.java

```diff
@@ -0,0 +1,140 @@
+ * file that was distributed with this source code.
+ */
+
+public class ParameterEqualsDefaultValueInspector extends BasePhpInspection {
+    private static final String message = "This parameter could be dropped, because the value is the same from default value.";
+
+    @NotNull
+    public final String getShortName() {
+        return "ParameterEqualsDefaultValueInspection";
+    }
+
+    @NotNull
+    @Override
+    public final PsiElementVisitor buildVisitor(@NotNull final ProblemsHolder problemsHolder, final boolean onTheFly) {
+        return new BasePhpElementVisitor() {
+            @Override
+            public void visitPhpFunctionCall(final FunctionReference reference) {
+                final PsiElement[] referenceParameters = reference.getParameters();
+
+                if (referenceParameters.length == 0) {
+                    return;
+                }
+
+                final Function function = (Function) reference.resolve();
+
+                if (function == null) {
+                    return;
+                }
+
+                final Parameter[] functionParameters = function.getParameters();
+
+                if (functionParameters.length == 0) {
+                    return;
+                }
+
+                PsiElement referenceParameterLower  = null;
+                final int  referenceParametersLimit = Math.min(referenceParameters.length, functionParameters.length) - 1;
+
+                for (int parameterIndex = referenceParametersLimit; parameterIndex >= 0; parameterIndex--) {
+                    final PhpExpression referenceParameter = (PhpExpression) referenceParameters[parameterIndex];
+                    final Parameter     functionParameter  = functionParameters[parameterIndex];
+
+                    if (referenceParameter.getType().equals(functionParameter.getType())) {
+                        final PsiElement functionParameterDefaultValue = functionParameter.getDefaultValue();
+
+                        if ((functionParameterDefaultValue != null) &&
+                            referenceParameter.getText().equals(functionParameterDefaultValue.getText())) {
+                            referenceParameterLower = referenceParameter;
+                            continue;
+                        }
+                    }
+
+                    break;
+                }
+
+                if (referenceParameterLower != null) {
+                    final PsiElement referenceParameterLowerPrev = ((PhpPsiElement) referenceParameterLower).getPrevPsiSibling();
+                    final PsiElement referenceParameterStart = (referenceParameterLowerPrev == null)
+                                                               ? referenceParameterLower
+                                                               : referenceParameterLowerPrev.getNextSibling();
+
+                    problemsHolder.registerProblem(problemsHolder.getManager().createProblemDescriptor(
```

Comment:
My logic was: 

* L69: The looping is done from last argument to first.
* L76: It will "accept as unused" if the type and text from argument is the same from default value;
* L78: It will store the "first unused argument" (the last we just know, because is the last parameter);
* L79: then will keep searching for "even deeper equality".

---

* L86: If we found some equality;
* L92: Then we register the problem...
* L93: highlighting the first equals argument...
* L94: to last argument on call.

---

* L87: backing to this line, we too store the "comma", because the fix will drop it too.
* L88: if we have no previous argument, then it should drop from first to last; if we have previous argument, it will drop from comma immediately after the previous argument (that will not be dropped) to last argument;